components/team-members-overview/team-members-overview.js

```diff
@@ -9,170 +9,90 @@ class TeamMembersOverview extends Component {
+      this.state.activeTopic = queryParam // eslint-disable-line react/no-direct-mutation-state
+      this.state.filteredTeam = this.filterTeamMembers(team, department) // eslint-disable-line react/no-direct-mutation-state
+    }
   }
 
-  setQueryParams(filter, index, active) {
-    const url = '/team/people?'
-    let newUrl = ''
+  setQueryParams(filter) {
+    window.history.replaceState(null, null, encodeURI(`/team/people?filter=${filter}`))
+  }
 
-    if (!active) {
-      newUrl = `filter=${filter[index].value}`
-    }
+  filterTeamMembers(team, role) {
+    return team
+      .filter(member => !member.hide)
+      .filter(member => this.hasSelectedDepartment(member, role))
+  }
 
-    window.history.replaceState(null, null, `${encodeURI(`${url}${newUrl}`)}`)
+  hasSelectedDepartment(member, role) {
+    if (role === 'Everyone')
+      return true
+    else {
+      return member.departments.some(department => department.title === role)
+    }
   }
 
-  handleClick() {
-    const { isFilterCollapsed } = this.state
-    this.setState({ isFilterCollapsed: !isFilterCollapsed })
+  getDepartments(data) {
+    const allDepartments = data
+      .map(item => item.departments.map(item => item.title))
+      .reduce((a, b) => a.concat(b), [])
+    const uniqueDepartments = [...new Set(allDepartments)]
+    uniqueDepartments.unshift('Everyone')
+
+    return uniqueDepartments
   }
 
   render() {
-    const { team, introText } = this.props
-    const { roles, locations, isFilterCollapsed } = this.state
-    const filteredTeam = filterTeam(team, this.state)
-    const buttonIcon = isFilterCollapsed ? 'arrowDown' : 'arrowUp'
-    const buttonClass = isFilterCollapsed ? 'arrow-down' : 'arrow-up'
-    const filtersContainerClass = isFilterCollapsed ? 'hide' : ''
+    const { introText } = this.props
+    const { activeTopic, departments, filteredTeam } = this.state
 
     return (
       <div className="filters">
-        <ButtonSecondary
-          onClick={this.handleClick}
-          classes={`btn-red-border vertical-spring ${buttonClass} filters-toggle`}
-          icon={buttonIcon}>
-          Filters
-        </ButtonSecondary>
-        <div className={`${filtersContainerClass} filters-container`}>
-          <Filter filter={roles} onFilter={this.onFilterHandler} />
-          <Filter filter={locations} onFilter={this.onFilterHandler} />
+        <div className="filters-container">
+          <Topics
+            keyword="Role"
+            activeTopic={activeTopic}
+            topics={departments}
+            onTopicChanged={this.changeTopicHandler} />
         </div>
 
         {introText && <p className="team-members-intro-text container">{introText}</p>}
         <ul className="team-members-overview container">
-          {filteredTeam.map((teamMember, index) => (
-            <TeamMember key={index} data={teamMember} />
+          {filteredTeam.map((member, index) => (
+            <TeamMember key={index} data={member} />
```

Comment:
you could write this as a one liner without the index parameter and use the name of the member as key:
 {filteredTeam.map(member => <TeamMember key={member.name} data={member} />)}